src/main/java/com/codeborne/selenide/appium/commands/AppiumScrollTo.java

```diff
@@ -0,0 +1,86 @@
+package com.codeborne.selenide.appium.commands;
+
+import com.codeborne.selenide.SelenideElement;
+import com.codeborne.selenide.commands.ScrollTo;
+import com.codeborne.selenide.impl.WebElementSource;
+import io.appium.java_client.AppiumDriver;
+import org.openqa.selenium.Dimension;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.interactions.PointerInput;
+import org.openqa.selenium.interactions.Sequence;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.util.Optional;
+
+import static com.codeborne.selenide.appium.WebdriverUnwrapper.cast;
+import static java.time.Duration.ofMillis;
+import static java.util.Collections.singletonList;
+
+
+@ParametersAreNonnullByDefault
+public class AppiumScrollTo extends ScrollTo {
+
+  private AppiumDriver appiumDriver;
+
+  @Override
+  @Nonnull
+  public SelenideElement execute(SelenideElement proxy, WebElementSource locator, @Nullable Object[] args) {
+    Optional<AppiumDriver> appiumDriverOptional = cast(locator.driver(), AppiumDriver.class);
+    if (!appiumDriverOptional.isPresent()) {
+      return super.execute(proxy, locator, args);
+    }
+
+    this.appiumDriver = appiumDriverOptional.get();
+
+    int currentSwipeCount = 0;
+    String previousPageSource = "";
+
+    while (isElementNotDisplayed(locator) && isNotEndOfPage(previousPageSource) && isLessThanMaxSwipeCount(currentSwipeCount)) {
+      previousPageSource = appiumDriver.getPageSource();
+      performScroll();
+      currentSwipeCount++;
+    }
+    return proxy;
+  }
+
+  private boolean isLessThanMaxSwipeCount(int currentSwipeCount) {
+    final int maxSwipeCount = 30;
+    return currentSwipeCount < maxSwipeCount;
+  }
+
+  private boolean isElementNotDisplayed(WebElementSource locator){
+    try{
+      return !locator.getWebElement().isDisplayed();
+    }
+    catch (NullPointerException | NoSuchElementException exception){
```

Comment:
Why ignore `NullPointerException`? Generally, it's not a good idea. We might receive a "real" NPE from Selenium or Selenide code. If we don't see it. we will never fix it.

Reply:
@asolntsev - In this case, we are getting null pointer exception when calling locator.getWebElement. If it is coming from selenium, is it not already handled when we call $(WebElement)

Reply:
@amuthansakthivel Can you share stacktrace of this NPE?

Reply:
@asolntsev - You are correct. I have now removed the NPE from try catch.

PS:  May be initially I handled NPE because there was actually a problem with my code which was refactored later.